R/spark_shell.R

```diff
@@ -7,6 +7,68 @@ shell_connection_validate_config <- function(config) {
   config
 }
 
+databricks_connection <- function(config, extensions) {
+  tryCatch({
+    gatewayPort <- as.numeric(SparkR:::callJStatic("com.databricks.backend.daemon.driver.RDriverLocal",
+                                                  "startSparklyr",
+                                                  # In Databricks notebooks, this variable is in the default namespace
+                                                  get("DATABRICKS_GUID", envir = .GlobalEnv),
+                                                  # startSparklyr will search & find proper JAR file
+                                                  system.file("java/", package = "sparklyr")))
+  }, error = function(err) {
+    stop(paste("Failed to start sparklyr backend:", err$message))
+  })
+
+  tryCatch({
+    gatewayInfo <- spark_connect_gateway(gatewayAddress = "localhost",
+                                        gatewayPort = gatewayPort,
+                                        sessionId = gatewayPort,
+                                        config = config,
+                                        isStarting = TRUE)
+  }, error = function(err) {
+    stop(paste("Failed to open connection to sparklyr gateway:", err$message))
+  })
+
+  output_file <- tempfile(fileext = "_spark.log")
+  error_file <- tempfile(fileext = "_spark.err")
+
+  tryCatch({
+    # set timeout for socket connection
+    timeout <- spark_config_value(config, "sparklyr.backend.timeout", 30 * 24 * 60 * 60)
+    backend <- socketConnection(host = "localhost",
+                                port = gatewayInfo$backendPort,
+                                server = FALSE,
+                                blocking = TRUE,
+                                open = "wb",
+                                timeout = timeout)
+  }, error = function(err) {
+    stop(paste("Failed to open connection to sparklyr backend:", err$message))
+  })
+
+  # create the databricks connection
+  sc <- structure(class = c("spark_connection", "spark_shell_connection"), list(
+    # spark_connection
+    master = "databricks",
```

Comment:
Would anything break on your side if we change `master="databricks"` into, for instance, `master="remote"`? Or does the infrastructure assumes this needs to be set to `databricks`?